runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -739,9 +739,9 @@ private void applyForEachAddress(BiConsumer<Long, ILogData> logDataProcessor) {
         while (nextRead <= logTail) {
             try {
                 final long start = nextRead;
-                final long stopNotIncluded = Math.min(start + batchReadSize, logTail + 1);
-                nextRead = stopNotIncluded;
-                final Map<Long, ILogData> range = getLogData(runtime, start, stopNotIncluded);
+                final long stopIncluded = Math.min(start + batchReadSize - 1, logTail);
```

Comment:
Maybe its better to use interval terminology here, perhaps LeftEndpoint and RightEndpoint?

Reply:
Done.